### PR TITLE
refactor(blocks): improve the configuration of more menu

### PR DESCRIPTION
**Pull request by fundon**
* caches more menu configuration
* adds `getMoreMenuConfig` method
* splits context and config files

BTW: currently, this can only be configured in `firstUpdated`, because it depends on `std`

### CODE REVIEW

**Comment by fundon** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8089?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8089** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8089?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#8077** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8077?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8072** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8072?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8070** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8070?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/blocks/src/root-block/configs/toolbar.ts [hunk 1]
```diff
@@ -42,3 +35,10 @@ export interface ToolbarMoreMenuConfig {
     groups: MenuItemGroup<T>[]
   ) => MenuItemGroup<T>[];
 }
+
+export function getMoreMenuConfig(std: BlockStdScope): ToolbarMoreMenuConfig {
+  return {
+    configure: <T extends MenuContext>(groups: MenuItemGroup<T>[]) => groups,
+    ...std?.spec.getConfig('affine:page')?.toolbarMoreMenu,
```
Comment (fourdim) [inline packages/blocks/src/root-block/configs/toolbar.ts:42]: `std?` here `?` is not required.

Comment (fourdim) [inline packages/blocks/src/root-block/configs/toolbar.ts:42]: `std?` here `?` is not required.

File: packages/blocks/src/attachment-block/components/context.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  get doc() {
+    return this.blockComponent.doc;
+  }
+
+  get host() {
+    return this.blockComponent.host;
+  }
+
+  get selectedBlockModels() {
+    return [this.blockComponent.model];
```
Comment (fourdim) [inline packages/blocks/src/attachment-block/components/context.ts:38]:
```suggestion
    if (this.blockComponent.model) return [this.blockComponent.model];
    return [];
```

File: packages/blocks/src/root-block/widgets/surface-ref-toolbar/config.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          const edgelessRootElement = editor?.view.getBlock(ctx.doc.root.id);
+          const surfaceRenderer = (
+            edgelessRootElement as EdgelessRootPreviewBlockComponent
+          )?.surface?.renderer;
+
+          edgelessToBlob(ctx.host, {
+            surfaceRefBlock: ctx.blockComponent,
+            surfaceRenderer,
+            edgelessElement: referencedModel,
+          })
```
Comment (fourdim) [inline packages/blocks/src/root-block/widgets/surface-ref-toolbar/config.ts:39]: `surfaceRender` can be `undefined`.

File: packages/blocks/src/root-block/widgets/surface-ref-toolbar/config.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
```
Comment (fourdim) [inline packages/blocks/src/root-block/widgets/surface-ref-toolbar/config.ts:65]: Also here.

File: packages/blocks/src/root-block/configs/toolbar.ts (hunk 1 above)
Comment (fundon) [reply packages/blocks/src/root-block/configs/toolbar.ts:42]: Be safe

**Review by fourdim — APPROVED** [review]

File: packages/blocks/src/attachment-block/components/context.ts (hunk 2 above)
Comment (fundon) [reply packages/blocks/src/attachment-block/components/context.ts]: Thanks for the comments.

**Pull request merged**